bench/workgen/validate_mirror_tables.py

```diff
@@ -0,0 +1,137 @@
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+
+# ------------------------------------------------------------------------------
+# Given a database directory generated by Workgen, find and compare the mirrored
+# data files. Returns zero on success if all mirror pairs contain the same data.
+# Returns a non-zero value on failure if any mirror pair contains a mismatch.
+#
+# Note that the wiredtiger/tools directory must be on your PYTHONPATH.
+# ------------------------------------------------------------------------------
+import os, pathlib, re, sys
+from contextlib import redirect_stdout
+from wt_tools_common import wiredtiger_open
+from wt_cmp_uri import wiredtiger_compare_uri
+
+# Print usage and exit with a failure status.
+def usage_exit():
+
+    print('Usage: python3 validate_mirror_tables database_dir')
+    print('  database_dir is a POSIX pathname to a WiredTiger home directory')
+    sys.exit(1)
+
+# Given a database directory, return all files with a '.wt' extension.
+def get_wiredtiger_db_files(dir):
+
+    files = [f for f in os.listdir(dir) if f.endswith('.wt')]
+    return files
+
+# Given a list of database files, return a list of mirrored uris.
+def get_mirrors(db_dir, db_files):
+
+    mirrors = []
+    db_files_remaining = set(db_files)
+
+    for uri in db_files:
+        if uri not in db_files_remaining:
+            continue
+        db_files_remaining.remove(uri)
+        mirror_uri = get_mirror_uri(db_dir, f'file:{uri}')
+        if mirror_uri is not None:
+            db_files_remaining.remove(mirror_uri)
+            mirrors.append([f'{db_dir}/file:{uri}',
+                            f'{db_dir}/file:{mirror_uri}'])
+
+    return mirrors
+
+# Get the mirror for the specified uri by examining the uri's metadata. Mirror names
+# are stored in the 'app_metadata' by workgen when mirroring is enabled. If the uri has
+# a mirror, the name of the mirror is returned. Otherwise, the function returns None.
+def get_mirror_uri(db_dir, uri):
+
+    connection = wiredtiger_open(db_dir, 'readonly')
+    session = connection.open_session()
+    c = session.open_cursor('metadata:', None, None)
+    metadata = c[uri]
+    c.close()
+
+    result=re.findall('app_metadata="([^"]*)"', metadata)
+    mirror = None
+
+    if result:
+        app_metadata = dict((a.strip(), b.strip())  
+            for a, b in (element.split('=')  
+                for element in result[0].split(',')))  
+
+        if app_metadata['workgen_dynamic_table'] == 'true':
+            mirror = app_metadata['workgen_table_mirror']
+            is_base = (app_metadata['workgen_base_table'] == 'true')
```

Comment:
It doesn't look like `is_base` is used after assignment.

Reply:
I missed that one, thanks! Removed in [cb52db0](https://github.com/wiredtiger/wiredtiger/pull/8865/commits/cb52db0db127d5b522d0e065730f54c8081a9ce7).